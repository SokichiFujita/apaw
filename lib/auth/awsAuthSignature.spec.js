const awsAuthSignature = require('./awsAuthSignature');

describe('awsAuthSignature', () => {
  it('should return a string that represents a signed v4 AWS request', () => {
    const authorization = awsAuthSignature({
      headers: {
        'Content-Encoding': 'amz-1.0',
        'X-Amz-Date': '20200308T123600Z',
      },
      key: 'key',
      longDate: '20200308T123600Z',
      path: '/',
      payload: 'payload',
      region: 'region',
      secret: 'secret',
      shortDate: '20200308',
    });
    expect(authorization).toMatchSnapshot();
  });
});
