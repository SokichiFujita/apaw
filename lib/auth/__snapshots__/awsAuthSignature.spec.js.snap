// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`awsAuthSignature should return a string that represents a signed v4 AWS request 1`] = `"AWS4-HMAC-SHA256 Credential=key/20200308/region/ProductAdvertisingAPI/aws4_request, SignedHeaders=content-encoding;x-amz-date, Signature=63a9da15fd23d41338445717369b139c7a615425cf36836a444ad464697cb7c7"`;
